earth_enterprise/src/fusion/autoingest/sysman/khSystemManager.cpp

```diff
@@ -72,23 +73,34 @@ khSystemManager::SetWantExit(void)
   }
 }
 
-void
-khSystemManager::SignalLoop(void)
+// when SIGHUP is given, reload systemrc
+void systemrc_reload(int signum)
 {
-  sigset_t waitset;
-  sigemptyset(&waitset);
-  sigaddset(&waitset, SIGINT);
-  sigaddset(&waitset, SIGTERM);
+    notify(NFY_WARN, "Recieved signal %s, Reloading systemrc...",
+           khNotifyLevelToString(static_cast<khNotifyLevel>(signum)).c_str());
+    Systemrc systemrc;
+    LoadSystemrc(systemrc);
+    uint32 logLevel = systemrc.logLevel;
+    notify(NFY_WARN, "system log level changed to: %s",
+           khNotifyLevelToString(static_cast<khNotifyLevel>(logLevel)).c_str());
+    setNotifyLevel(static_cast<khNotifyLevel>(logLevel));
+}
 
+void handleExitSignals(int signum)
+{
+    notify(NFY_WARN, "Recieved signal %s, Exiting...",
+           khNotifyLevelToString(static_cast<khNotifyLevel>(signum)).c_str());
+    theSystemManager.SetWantExit();
+}
 
-  int sig;
-  sigwait(&waitset, &sig);
-  if (sig > 0) {
-    notify(NFY_NOTICE, "Received signal %d. Exiting ...", sig);
-    SetWantExit();
-  } else {
-    notify(NFY_NOTICE, "Received signal %d. Ignoring ...", sig);
-  }
+
+void
+khSystemManager::SignalLoop(void)
+{
+  signal(SIGINT,handleExitSignals);
+  signal(SIGTERM,handleExitSignals);
+  signal(SIGHUP,systemrc_reload);
+  while(1);
```

Comment:
I don't think you need this while here. Now that you've changed the way the system handler works this can just exit, I think. Signals will still be sent to the appropriate handlers.